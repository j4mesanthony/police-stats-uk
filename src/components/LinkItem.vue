<script setup lang="ts">
const props = defineProps({
    small: {
        type: Boolean,
        default: false
    },

    large: {
        type: Boolean,
        default: false
    },

    underline: {
        type: Boolean,
        default: true
    },

    isActive: {
        type: Boolean,
        default: false
    },

    icon: {
        type: String,
        default: ''
    },
});
</script>

<template>
    <a class="cursor-pointer text-slate-800 dark:hover:text-slate-300" :class="[ props.isActive ? 'dark:text-cyan-400' : 'dark:text-slate-400', props.small ? 'text-sm':'', props.large ? 'text-lg':'', props.underline ? 'underline':'' ]">
        <!-- TODO: set icon font size via prop -->
        <span class="relative material-symbols-outlined top-[4px] mr-1" style="font-size: 18px;">{{ props.icon }}</span><slot></slot>
    </a>
</template>