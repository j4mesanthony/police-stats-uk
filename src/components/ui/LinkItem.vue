<script setup lang="ts">
const props = defineProps({
    small: {
        type: Boolean,
        default: false
    },

    large: {
        type: Boolean,
        default: false
    },

    underline: {
        type: Boolean,
        default: true
    },

    isActive: {
        type: Boolean,
        default: false
    },

    icon: {
        type: String,
        default: ''
    },

    iconSize: {
        type: String,
        default: ''
    },
});
</script>

<template>
    <a class="cursor-pointer text-slate-300 hover:text-slate-300" :class="[ props.isActive ? 'text-cyan-400' : 'text-slate-400', props.small ? 'text-sm':'', props.large ? 'text-lg':'', props.underline ? 'underline':'' ]">
        <span class="relative material-symbols-outlined top-[4px] mr-1" :style="{ 'fontSize': props.iconSize }">{{ props.icon }}</span><slot></slot>
    </a>
</template>