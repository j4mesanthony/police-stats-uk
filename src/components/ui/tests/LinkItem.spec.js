import { describe, it, expect } from 'vitest';
import LinkItem from '../LinkItem.vue';
import { createApp } from 'vue';
import { renderToString } from '@vue/server-renderer';

describe('LinkItem', () => { 
    it('should not regress (snapshot test)', async () => {
        const app = createApp(LinkItem);
        const string = await renderToString(app);
        expect(string).toMatchSnapshot();
    })
 })