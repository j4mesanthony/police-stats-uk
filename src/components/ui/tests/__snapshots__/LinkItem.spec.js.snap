// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LinkItem > should not regress (snapshot test) 1`] = `"<a class=\\"cursor-pointer text-slate-300 hover:text-slate-300 text-slate-400 underline\\"><span class=\\"relative material-symbols-outlined top-[4px] mr-1\\" style=\\"font-size:;\\"></span><!--[--><!--]--></a>"`;
